src/standalone/Soil/energy_hydrology.jl

```diff
@@ -645,7 +645,7 @@ function ClimaLand.source!(
     _ρ_i = FT(LP.ρ_cloud_ice(earth_param_set))
     Δz = model.domain.fields.Δz # center face distance
 
-    # Wrap hydrology and earth parameters in one struct to avoid type inference failure
+    # Wrap hydrology and earth parameters in one struct to avoid type inference failure. This is allocating! We should remove it.
```

Comment:
This was something @juliasloan25 implemented because otherwise we had the GPU error on central. it's a climacore bug if I understood correctly - we couldnt broadcast over hydrology_cm and earth_param_set separately.

but maybe we can refactor the function - hydrology_cm and earth_param_set are packaged inside of model.parameters already

Reply:
Yes, I didn't tackle it because it leads to inference issues.